mongodbatlas/data_source_mongodbatlas_search_index.go

```diff
@@ -140,21 +153,34 @@ func dataSourceMongoDBAtlasSearchIndexRead(ctx context.Context, d *schema.Resour
 		return diag.Errorf("error setting `searchAnalyzer` for search index (%s): %s", d.Id(), err)
 	}
 
-	if err := d.Set("mappings_dynamic", searchIndex.Mappings.Dynamic); err != nil {
-		return diag.Errorf("error setting `mappings_dynamic` for search index (%s): %s", d.Id(), err)
-	}
-
 	if err := d.Set("synonyms", flattenSearchIndexSynonyms(searchIndex.Synonyms)); err != nil {
 		return diag.Errorf("error setting `synonyms` for search index (%s): %s", d.Id(), err)
 	}
 
-	if len(searchIndex.Mappings.Fields) > 0 {
-		searchIndexMappingFields, err := marshalSearchIndex(searchIndex.Mappings.Fields)
+	if searchIndex.Mappings != nil {
```

Comment:
we are changing this if statement - do you foresee any potential breaking change?

Reply:
I tried different tests (automatic and manual) and couldn't find any problem. In current indexes searchInde.Mapping is not nil, this statement is false only for (new) vector indexes